src/Ormolu/Fixity/Imports.hs

```diff
@@ -71,3 +74,18 @@ ieToOccNames = \case
   IEThingAll _ (L _ x) -> [occName x] -- TODO not quite correct, but how to do better?
   IEThingWith _ (L _ x) _ xs -> occName x : fmap (occName . unLoc) xs
   _ -> []
+
+-- | Apply given module re-exports.
+applyModuleReexports :: ModuleReexports -> [FixityImport] -> [FixityImport]
+applyModuleReexports (ModuleReexports reexports) imports = imports >>= expand
+  where
+    expand i = do
+      case Map.lookup (fimportModule i) reexports of
+        Nothing -> pure i
+        Just exports ->
+          let exportToImport mmodule =
+                i
+                  { fimportPackage = Nothing,
+                    fimportModule = mmodule
+                  }
+           in NE.toList exports >>= expand . exportToImport
```

Comment:
Do we have/want a dedicated test that module re-exports of re-exports work as expected?

Reply:
Indeed we have! https://github.com/tweag/ormolu/blob/mkarpov%40issue-1017/tests/Ormolu/FixitySpec.hs#L232-L246